layers/timeline_semaphore.c

```diff
@@ -2088,7 +2269,49 @@ static VkResult timeline_CreateDevice(
     /* Advance the link info for the next element on the chain */
     chain_info->u.pLayerInfo = chain_info->u.pLayerInfo->pNext;
 
-    VkResult result = fpCreateDevice(physicalDevice, pCreateInfo, pAllocator, pDevice);
+    /* Clear timelineSemaphore before calling icd CreateDevice since icd possibly doesn't support it.
+     * Even icd supports it, do not enable it when this layer is enabled.
+     */
+    const VkPhysicalDeviceVulkan12Features *vulkan12_features =
+        vk_find_struct_const(pCreateInfo->pNext, PHYSICAL_DEVICE_VULKAN_1_2_FEATURES);
+    const VkPhysicalDeviceTimelineSemaphoreFeaturesKHR *timeline_features =
+        vk_find_struct_const(pCreateInfo->pNext, PHYSICAL_DEVICE_TIMELINE_SEMAPHORE_FEATURES_KHR);
+
+    VkDeviceCreateInfo *pNewCreateInfo = NULL;
+    void *pNewNext = NULL;
+
+    if (vulkan12_features || timeline_features) {
+        pNewCreateInfo = vk_alloc(pAllocator, sizeof(VkDeviceCreateInfo), 8, VK_SYSTEM_ALLOCATION_SCOPE_DEVICE);
```

Comment:
I guess if you handled `VK_STRUCTURE_TYPE_CREATE_DEVICE_INFO` in `vk_device_create_info_type_size` you could replace this chunk with : 
```
pNewCreateInfo = clone_device_create_info_pnext(pAllocator, pCreateInfo);
```

Reply:
Ok.